src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -361,9 +362,17 @@ HRESULT STDMETHODCALLTYPE CorProfiler::JITCompilationStarted(
                              target.signature);
       }
 
-      // replace with a call to the instrumentation wrapper
       const auto original_argument = pInstr->m_Arg32;
+
+      // insert the opcode and signature token as
+      // additional arguments for the wrapper method
+      ILRewriterWrapper rewriter_wrapper(&rewriter);
+      rewriter_wrapper.SetILPosition(pInstr);
+      rewriter_wrapper.LoadInt32(pInstr->m_opcode);
```

Comment:
This is where we insert the original call's opcode value into the execution stack before replacing the method call.